packages/core/test-helpers/core-test-helpers-kbn-server/src/create_serverless_root.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { defaultsDeep } from 'lodash';
+import { Cluster } from '@kbn/es';
+import Path from 'path';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { ToolingLog } from '@kbn/tooling-log';
+import execa from 'execa';
+import { CliArgs } from '@kbn/config';
+import { createRoot, type TestElasticsearchUtils, type TestKibanaUtils } from './create_root';
+
+export type TestServerlessESUtils = Pick<TestElasticsearchUtils, 'stop' | 'es'>;
+export type TestServerlessKibanaUtils = TestKibanaUtils;
+export interface TestServerlessUtils {
+  startES: () => Promise<TestServerlessESUtils>;
+  startKibana: (abortSignal?: AbortSignal) => Promise<TestServerlessKibanaUtils>;
+}
+
+export function createTestServerlessInstances({
```

Comment:
Seemed easiest to just create a separate function rather than adding a flag to the existing `createTestServers` logic.

However, we mirror the existing API as closely as possible so that these helpers can be easily substituted.

Reply:
you said the quiet part out loud 🤫

Reply:
but in all seriousness: if you have a better idea for a name I'll gladly accept 😄

Reply:
Na, naming is hard, it's fine. We can rename later if we really want so.